Sem. 03/Solutions/Task01.cpp

```diff
@@ -0,0 +1,183 @@
+
+            for (int j = i + 1; j < arraySize; ++j) {
+
+                if (isSecondStringLarger(deliveryArray[j].address, deliveryArray[minIndex].address)) {
+                    minIndex = j;
+                }
+
+            }
+
+            if (minIndex != i) {
+                swapDeliveries(deliveryArray[i], deliveryArray[minIndex]);
+            }
+        }
+    }
+
+    void initializeDelivery(Delivery& delivery, const double price, const char* address) {
+        delivery.price = price;
+        assignAddress(delivery, address);
+    }
+
+    void writeDeliveryArrayToFile(std::ofstream& out, const Delivery* deliveryArray, const size_t arraySize) {
+        out.write((const char*)deliveryArray, sizeof(Delivery) * arraySize);
+    }
+
+    void writeSortedDeliveryArrayToFile(std::ofstream& out, Delivery* deliveryArray, const size_t arraySize) {
+        selectionSort(deliveryArray, arraySize);
+        out.write((const char*)deliveryArray, sizeof(Delivery) * arraySize);
+    }
+
+    void readDeliveryArrayFromFile(std::ifstream& in, Delivery*& deliveryArray, size_t& arraySize) {
+        arraySize = getFileSize(in) / sizeof(Delivery);
+        deliveryArray = new Delivery[arraySize];
+        in.read((char*)deliveryArray, sizeof(Delivery) * arraySize);
+    }
+
+    void printDeliveryArray(const Delivery* deliveryArray, const size_t arraySize) {
+        
+        for (int i = 0; i < arraySize; ++i) {
+            std::cout << deliveryArray[i].price << " " << deliveryArray[i].address << '\n';
+        }
+    }
+
+}
+
+int main()
+{
+    size_t n = 0;
+    double price = 0;
+    char* address = new char[ADDRESS_MAX_LENGTH];
+
+    Delivery* deliveryArrayFromUnsortedFile = nullptr;
+    size_t unsortedArraySize = 0;
+    Delivery* deliveryArrayFromSortedFile = nullptr;
+    size_t sortedArraySize = 0;
+
+    std::cin >> n;
+
+    Delivery* deliveries = new Delivery[n];
+
+    for (int i = 0; i < n; ++i) {
+        std::cin >> price;
+        std::cin.ignore();
+        std::cin.getline(address, ADDRESS_MAX_LENGTH);
+        deliveryFunctions::initializeDelivery(deliveries[i], price, address);
+    }
+
+    using namespace deliveryFunctions;
+
+    std::ofstream out1(UNSORTED_FILE_NAME, std::ios::binary);
+    std::ofstream out2(SORTED_FILE_NAME, std::ios::binary);
+
+    if (!out1.is_open() || !out2.is_open()) {
+        std::cerr << ERROR_MESSAGE;
+        return 1;
+    }
+
+    writeDeliveryArrayToFile(out1, deliveries, n);
+
+    writeSortedDeliveryArrayToFile(out2, deliveries, n);
+
+    out1.close();
+    out2.close();
+
+    std::ifstream in1(UNSORTED_FILE_NAME, std::ios::binary);
+    std::ifstream in2(SORTED_FILE_NAME, std::ios::binary);
+
+    if (!in1.is_open() || !in2.is_open()) {
+        std::cerr << ERROR_MESSAGE;
+        return 1;
+    }
+
+    readDeliveryArrayFromFile(in1, deliveryArrayFromUnsortedFile, unsortedArraySize);
+    readDeliveryArrayFromFile(in2, deliveryArrayFromSortedFile, sortedArraySize);
```

Comment:
same comment as above: separate the streams and close them before printing results